### PR TITLE
C++ API parity: MaxUnpool3d

### PR DESCRIPTION
**Pull request by pbelevich**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27060 C++ API parity: LogSigmoid
* #27059 C++ API parity: LeakyReLU
* #27038 C++ API parity: Hardtanh
* #27035 C++ API parity: Hardshrink
* #27028 C++ API parity: ELU
* **#27027 C++ API parity: MaxUnpool3d**

Differential Revision: [D17682402](https://our.internmc.facebook.com/intern/diff/D17682402)

### CODE REVIEW

File: test/cpp/api/modules.cpp [hunk 1]
```diff
@@ -605,6 +605,61 @@ TEST_F(ModulesTest, MaxUnpool2d) {
...
+      {{ 0,  0,  0,  0},
+       { 0,  0,  0,  0},
+       { 0,  0,  0,  0},
+       { 0,  0,  0,  0}},
+      {{ 0,  0,  0,  0},
+       { 0, 53,  0, 55},
+       { 0,  0,  0,  0},
+       { 0, 61,  0, 63}}}}}, torch::kFloat)));
+  ASSERT_EQ(y.sizes(), torch::IntArrayRef({1, 1, 4, 4, 4}));
+}
```
Comment (yf225) [inline test/cpp/api/modules.cpp]: It would be great to add a test case for MaxPool3d -> MaxUnpool3d, similar to the 1d and 2d cases.

Comment (pbelevich) [reply test/cpp/api/modules.cpp]: done

**Review by yf225 — APPROVED** [review]
Thanks @pbelevich !

**Comment by yf225** [comment]
@pbelevich Sorry I just realized that we didn't add a test for `F::max_unpool3d` in `test/cpp/api/functional.cpp`, curious would you like to add one? Thanks!

**Pull request closed without merging**